test/e2e/framework/conformance/conformance.go

```diff
@@ -0,0 +1,940 @@
+	//
+	// Also checks the informers. This is a bit redundant after read-only steps,
+	// but then it's also fast and thus can be done more often than strictly necessary.
+	// Collected informer events get reset, so each verify call must match
+	// events since the previous one.
+
+	verify := func(what string, expected, actual *unstructured.Unstructured, haveExpectedEvents gtypes.GomegaMatcher) {
+		ginkgo.GinkgoHelper()
+
+		// This captures several different failures before handing them to Ginkgo.
+		var failures gomegaFailures
+
+		if expected != nil {
+			if tc.VerifyContent() {
+				diff := compareObjects(expected, actual)
+				failures.Add(fmt.Sprintf("%s: unexpected actual object (- expected, + actual):\n%s", what, diff))
+			} else {
+				failures.G().Expect(actual.GetName()).Should(gomega.Equal(expected.GetName()), "%s: name in returned object", what)
+				failures.G().Expect(actual.GetNamespace()).Should(gomega.Equal(expected.GetNamespace()), "%s: namespace in returned object", what)
+			}
+			failures.G().Expect(actual).To(apimachineryutils.HaveValidResourceVersion())
+		}
+
+		// Abort checking now if there were failures, otherwise we just risk timining out slowly.
+		failures.Check()
+
+		failures.G().Eventually(ctx, listResource).Should(matchObject(expected), "list of resources from informer cache after %s", what)
+		if haveExpectedEvents != nil {
+			// Even if the cache is up-to-date we still need to wait for event delivery.
+			failures.G().Eventually(resourceEvents.list).
+				WithTimeout(5*time.Second).
+				Should(haveExpectedEvents, "list of resource informer events after %s", what)
+		}
+		nextEvents(&resourceRV, resourceEvents.reset())
+		if listNamespace != nil {
+			failures.G().Eventually(ctx, listNamespace).Should(matchObject(expected), "list of namespace from informer cache after %s", what)
+			if haveExpectedEvents != nil {
+				failures.G().Eventually(namespaceEvents.list).
+					WithTimeout(5*time.Second).
+					Should(haveExpectedEvents, "list of namespace informer events after %s", what)
+			}
+			nextEvents(&namespaceRV, namespaceEvents.reset())
+		}
+		failures.Check()
+	}
+
+	// Create the initial resource.
+	ginkgo.By(fmt.Sprintf("Creating:\n%s", format.Object(desiredInitialObject, 1)))
+	existingObject, err := client.Create(ctx, desiredInitialObject, metav1.CreateOptions{FieldValidation: "Strict"})
+	framework.ExpectNoError(err, "create initial %s", resourceType)
+	ginkgo.DeferCleanup(func(ctx context.Context) {
+		// Always clean up.
+		err = client.Delete(ctx, desiredInitialObject.GetName(), metav1.DeleteOptions{})
+		if apierrors.IsNotFound(err) {
+			return
+		}
+		framework.ExpectNoError(err, "delete %s", resourceType)
+		ensureNotFound(ctx, getResource)
+	})
+	verify("create", desiredInitialObject, existingObject,
+		// Initial creation of the object followed by some optional updates by cluster components.
+		matchEvents(existingObject, "^add,(update,)*$"),
+	)
```

Comment:
save `createdResourceVersion := existingObject.GetResourceVersion()` here, and compare specifically with the resource version after `verify("update"...` to make sure the resource version *increased*